assembly/src/parsers/ast/field_ops.rs

```diff
@@ -1,12 +1,96 @@
 use super::{
-    super::parse_bit_len_param, super::parse_element_param, AssemblyError, Instruction, Node,
-    Token, Vec,
+    super::parse_element_param,
+    AssemblyError,
+    Instruction::*,
+    Node::{self, Instruction},
+    Token,
 };
-use crate::validate_operation;
 
+// INSTRUCTION PARSERS
+// ================================================================================================
+
+/// Returns `Add` instruction node if no immediate value is provided or `AddImm` instruction
+/// node otherwise.
+///
+/// # Errors
+/// Returns an error if the instruction token has invalid param or more than one param.
+pub fn parse_add(op: &Token) -> Result<Node, AssemblyError> {
+    debug_assert_eq!(op.parts()[0], "add");
```

Comment:
Should we use the `to_string` implementations of `Display` here? This way we will validate not only the correctness of the token, but also its display implementation